ArcGISRuntimeSDKQt_QMLSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.qml

```diff
@@ -0,0 +1,423 @@
+            } else {
+                return "`%1` %2 `%3`".arg(networkAttribute.name).arg(operatorAsString).arg(otherNetworkAttribute? otherNetworkAttribute.name : expression.value)
+            }
+        case Enums.UtilityTraceConditionTypUtilityCategoryComparisone:
+            return "`%1` %2".arg(networkAttribute.category.name).arg(networkAttribute.comparisonOperator === Enums.UtilityCategoryComparisonOperatorExists ? "Exists" : "DoesNotExist");
+        case Enums.UtilityTraceConditionTypeUtilityTraceAndCondition:
+            return "`%1` AND\n `%3`".arg(expressionToString(expression.leftExpression)).arg(expressionToString(expression.rightExpression));
+        case Enums.UtilityTraceConditionTypeUtilityTraceOrCondition:
+            return "`%1` OR\n `%3`".arg(expressionToString(expression.leftExpression)).arg(expressionToString(expression.rightExpression));
+        }
+    }
+
+    function comparisonOperatorToString(comparisonOperator) {
+        switch (comparisonOperator) {
+        case Enums.UtilityAttributeComparisonOperatorEqual:
+            return "Equal";
+        case Enums.UtilityAttributeComparisonOperatorNotEqual:
+            return "NotEqual";
+        case Enums.UtilityAttributeComparisonOperatorGreaterThan:
+            return "GreaterThan";
+        case Enums.UtilityAttributeComparisonOperatorGreaterThanEqual:
+            return "GreaterThanEqual";
+        case Enums.UtilityAttributeComparisonOperatorLessThan:
+            return "LessThan";
+        case Enums.UtilityAttributeComparisonOperatorLessThanEqual:
+            return "LessThanEqual";
+        case Enums.UtilityAttributeComparisonOperatorIncludesTheValues:
+            return "IncludesTheValues";
+        case Enums.UtilityAttributeComparisonOperatorDoesNotIncludeTheValues:
+            return "DoesNotIncludeTheValues";
+        case Enums.UtilityAttributeComparisonOperatorIncludesAny:
+            return "IncludesAny";
+        case Enums.UtilityAttributeComparisonOperatorDoesNotIncludeAny:
+            return "DoesNotIncludeAny";
+        default:
+            return null;
+        }
+    }
+
+    function addCondition() {
+        // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator.
+
+        if (traceConfiguration === null) {
+            traceConfiguration = ArcGISRuntimeEnvironment.createObject("UtilityTraceConfiguration");
+        }
+        if (traceConfiguration.traversability === null) {
+            traceConfiguration.traversability = ArcGISRuntimeEnvironment.createObject("UtilityTraversability");
+        }
+
+        const selectedNetworkAttribute = networkDefinition.networkAttribute(networkAttributeComboBox.currentText);
+        const data = convertToDataType(valueSelectionComboBox.visible ? valueSelectionComboBox.currentIndex : inputTextField.text, selectedNetworkAttribute.dataType)
+
+        if (isNaN(data)) {
```

Comment:
This is a bit confusing to me. This is the result of `convertToDataType`, and I would not think that a "type" could be `NAN`. I would suggest a minor cleanup here to clarify this.

Reply:
Good point. In the case no input was entered, it would end up returning 'nan' so this was a quick fix for that. I think my change is a better approach to a situation with invalid input.